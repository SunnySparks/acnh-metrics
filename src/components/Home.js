import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import '../App.css';
import { NavLink } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getVillagerLength } from '../redux/villagers/villagers';
import { getFossilLength } from '../redux/fossils/fossils';
import { getArtLength } from '../redux/art/art';

const Home = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getVillagerLength());
    dispatch(getFossilLength());
    dispatch(getArtLength());
  }, []);

  const villagers = useSelector((state) => state.villagersReducer);
  const fossils = useSelector((state) => state.fossilsReducer);
  const artsies = useSelector((state) => state.artsReducer);
  if (villagers.length < 1) {
    return (
      <div className="container p-4">
        <div className="row">
          <div className="col-sm-4" />
          <div className="col-sm-8">
            <h1>
              Loading
            </h1>
          </div>
        </div>
      </div>
    );
  }
  return (
    <nav className="navBar navbar-expand-lg p-4 border border-bottom bg-white">
      <div className="container-fluid">
        <div>
          <div>
            <div className="d-flex align-items-center">
              <div className="ms-5 logo-nav">
                <span className="me-4">
                  image
                </span>
                <span className="title-nav">title nav</span>
              </div>
              <div className="mb-1">|</div>
              <div>
                <NavLink
                  className="nav-link"
                  to="/villagers"
                >
                  <h2 className="links-nav">
                    Villagers:
                  </h2>
                  {villagers}
                  elements. Visible: 25.
                </NavLink>
              </div>
              <div className="mb-1">|</div>
              <div>
                <NavLink
                  className="nav-link"
                  to="/fossils"
                >
                  <h2 className="links-nav">
                    Fossils:
                  </h2>
                  {fossils}
                  elements.
                </NavLink>
              </div>
              <div className="mb-1">|</div>
              <div>
                <NavLink
                  className="nav-link"
                  to="/art"
                >
                  <h2 className="links-nav">
                    Works of Art:
                  </h2>
                  {artsies}
                  elements.
                </NavLink>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Home;
