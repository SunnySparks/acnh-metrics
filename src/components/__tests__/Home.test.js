import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Home from '../Home';

describe('Home component', () => {
  test('should render Home component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Home />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
