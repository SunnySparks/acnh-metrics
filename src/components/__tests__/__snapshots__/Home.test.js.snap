// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render Home component correctly 1`] = `
<div
  className="container"
>
  <div
    className="row head-banner headbanner pb-3"
  >
    <div
      className="col p-5 leafcont"
    />
    <div
      className="col p-5"
    >
      <h1>
        ACNH statistics
      </h1>
      507 Elements
    </div>
  </div>
  <div
    className="row text-white separatorBG lato pt-2"
  >
    <p>
      Categories
    </p>
  </div>
  <div
    className="row"
  >
    <div
      className="col category p-4 bg-1"
    >
      <NavLink
        className="nav-link"
        to="/villagers"
      >
        <h3
          className="links-nav"
        >
          Villagers:
        </h3>
      </NavLink>
      391 elements. Visible: 24.
    </div>
    <div
      className="col category p-4 bg-2"
    >
      <NavLink
        className="nav-link"
        to="/fossils"
      >
        <h3
          className="links-nav"
        >
          Fossils:
        </h3>
      </NavLink>
      73 elements.
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col category halfCont p-4 bg-2"
    >
      <NavLink
        className="nav-link"
        to="/art"
      >
        <h3
          className="links-nav"
        >
          Works of Art:
        </h3>
      </NavLink>
      43 elements.
    </div>
  </div>
</div>
`;
